model/host/db.go

```diff
@@ -1351,3 +1353,11 @@ func FindUnexpirableRunning() ([]Host, error) {
 	}
 	return hosts, db.FindAllQ(Collection, db.Query(q), &hosts)
 }
+
+// FindOneByPersistentDNSName returns hosts that have a matching persistent DNS
+// name.
+func FindOneByPersistentDNSName(ctx context.Context, dnsName string) (*Host, error) {
+	return FindOne(ctx, bson.M{
+		PersistentDNSNameKey: dnsName,
```

Comment:
should we add an index for this or other queries added here?

Reply:
There's already an index in staging on this key, I was just waiting for an LGTM to duplicate it in prod.

There's no other new queries being introduced in this PR, just updates by host ID.